src/compiler/scanner.ts

```diff
@@ -1353,7 +1353,8 @@ export function createScanner(languageVersion: ScriptTarget,
                 start = pos;
                 continue;
             }
-            if (isLineBreak(ch) && !jsxAttributeString) {
+
+            if ((isLineBreak(ch) && ch !== CharacterCodes.lineSeparator && ch !== CharacterCodes.paragraphSeparator) && !jsxAttributeString) {

```

Comment:
Perhaps at this point we should just check for the two valid line breaks rather than checking `isLineBreak` and then checking that it wasn't actually these two?

Reply:
> I'm trying to get a grasp of this and here's what [esbuild does](https://jakebailey.dev/esbuild-playground/#v2=Ly8gQGZpbGVuYW1lOiBjb25maWcuanNvbgp7CiAgImVudHJ5UG9pbnRzIjogWyIvaW5kZXgudHMiXSwKICAiZm9ybWF0IjogImVzbSIKfQoKLy8gQGZpbGVuYW1lOiBpbmRleC50cwovLyBTdHJpbmdzIGNvbnRhaW5pbmcgdW5lc2NhcGVkIGxpbmUgLyBwYXJhZ3JhcGggc2VwYXJhdG9ycwovLyBVc2luZyBib3RoIHNpbmdsZSBxdW90ZXMsIGRvdWJsZSBxdW90ZXMgYW5kIHRlbXBsYXRlIGxpdGVyYWxzCgpleHBvcnQgdmFyIHN0cmluZ0NvbnRhaW5pbmdVbmVzY2FwZWRMaW5lU2VwYXJhdG9yMSA9ICLigKhTVFJJTkdfQ09OVEVOVOKAqCI7CmV4cG9ydCB2YXIgc3RyaW5nQ29udGFpbmluZ1VuZXNjYXBlZFBhcmFncmFwaFNlcGFyYXRvcjEgPSAi4oCpU1RSSU5HX0NPTlRFTlTigKkiOwoKCmV4cG9ydCB2YXIgc3RyaW5nQ29udGFpbmluZ1VuZXNjYXBlZExpbmVTZXBhcmF0b3IyID0gJ-KAqFNUUklOR19DT05URU5U4oCoJzsKZXhwb3J0IHZhciBzdHJpbmdDb250YWluaW5nVW5lc2NhcGVkUGFyYWdyYXBoU2VwYXJhdG9yMiA9ICfigKlTVFJJTkdfQ09OVEVOVOKAqSc7CgoKZXhwb3J0IHZhciBzdHJpbmdDb250YWluaW5nVW5lc2NhcGVkTGluZVNlcGFyYXRvcjMgPSBg4oCoU1RSSU5HX0NPTlRFTlTigKhgOwpleHBvcnQgdmFyIHN0cmluZ0NvbnRhaW5pbmdVbmVzY2FwZWRQYXJhZ3JhcGhTZXBhcmF0b3IzID0gYOKAqVNUUklOR19DT05URU5U4oCpYDsKCi8vIEFycmF5IG9mIHVuZXNjYXBlZCBsaW5lIC8gcGFyYWdyYXBoIHNlcGFyYXRvcnMKCmV4cG9ydCB2YXIgYXJyID0gWwogICAgIuKAqeKAqFNUUklOR19DT05URU5U4oCo4oCpIiwKICAgICLigKjigKnigKlTVFJJTkdfQ09OVEVOVOKAqeKAqeKAqCIsCiAgICAiU1RSSU5HX0NPTlRFTlTigKkiLAogICAgIuKAqFNUUklOR19DT05URU5UIiwKICAgIGBc4oCpYCwKICAgICfigKgnCl07).
> 
> So, I think this is equivalent?


@jakebailey at a first glance, the modifications made in this PR should be equivalent with what esbuild does. Basically, the modifications allow for unescaped paragraph and line separators (U+2028 and U+2029 unicode characters) to be contained in string literals, as specified by the [JSON-superset](https://github.com/tc39/proposal-json-superset) ES 2019 proposal.

Reply:
> Should `tests/cases/compiler/sourceMap-LineBreaks.ts` have changed? My impression is that it includes these characters, but, nothing happens?

@jakebailey the baselines related to the test located in the `tests/cases/compiler/sourceMap-LineBreaks.ts` file haven't changed because the modifications made in this PR target the behavior of the two characters (U+2028 and U+2029) only if they appear unescaped in string literals (not outside of string literals, or inside string literals but escaped). 


As I can see from the test content, no unescaped U+2028/U+2029 chars appear in string literals:
`
// @target: ES3
// @sourcemap: true
var endsWithlineSeparator = 10;  var endsWithParagraphSeparator = 10;  var endsWithNextLine = 1;var endsWithLineFeed = 1;
var endsWithCarriageReturnLineFeed = 1;
var endsWithCarriageReturn = 1;
var endsWithLineFeedCarriageReturn = 1;

var endsWithLineFeedCarriageReturnLineFeed = 1;

var stringLiteralWithLineFeed = "line 1\
line 2";
var stringLiteralWithCarriageReturnLineFeed = "line 1\
line 2";
var stringLiteralWithCarriageReturn = "line 1\
line 2";

var stringLiteralWithLineSeparator = "line 1\ line 2"; var stringLiteralWithParagraphSeparator = "line 1\ line 2"; var stringLiteralWithNextLine = "line 1\line 2";
`

Reply:
> Perhaps at this point we should just check for the two valid line breaks rather than checking `isLineBreak` and then checking that it wasn't actually these two?

@jakebailey yeah, that sounds like a good idea. 

I have made the modification you've asked. 

My initial thought was that it would be more natural to exclude the two characters in order to treat the `isLineBreakfunction` as a black box. 

In that way, we don't actually care about what is actually a "line break", we just care about the fact that we want to ignore paragraph separators and line separators from being "line breaks". 

Hypothetically, if a brand new character would be considered a line break (and so the implementation of the `isLineBreak` function would change) we will still be covered without needing any code modifications (because we just exclude the line break and paragraph separator no matter what new characters are added as line breaks).

However, this is very unlikely to happen at this point, so we should focus more on the simplicity of the code. 

Please let me know if my point of view makes any sense. 

Thanks!

Reply:
Yeah, I'm torn between the two methods. We could just let it be the way it was.